### ISSUE TITLE
Question about dBF/dq2

### ISSUE THREAD

**Issue by gkaratha** [flav-io/flavio#229]
Dear experts,

My name is George Karathanasis and I am working on a measurement of dBF/dq2 for the B->mmK decay (for the CMS experiment). The last step of the analysis is to compare our measurements with several theoretical models. Therefore, we are thinking to use your package (with the proper citations of course). Now my question is the following. Reading the documentation I understood that the code is as simple as this[*]. I also used a different package named "EOS". Comparing the two I see large differences (see attached). So the 1st part of my question do you think my code [*] is wrong? If not is the difference between Flav-io and EOS expected? Thanks a lot in advance

Best regards,
George Karathanasis

[*]  import flavio as fl

BFtotal=0;
dBF=[]
for i in range(66):
  if i==0: q2_min=0.1
  else: q2_min=float(i)*(22.0-0.1)/66.0
  q2_max=(float(i)+1)*(22.0-0.1)/66.0
  print("bin",i,"min",q2_min,"max",q2_max,"dBF=",fl.sm_prediction('<dBR/dq2>(B+->Kmumu)',q2min=q2_min,q2max=q2_max))
  BFtotal+=fl.sm_prediction('<dBR/dq2>(B+->Kmumu)',q2min=q2_min,q2max=q2_max)*(q2_max-q2_min)
  dBF.append(fl.sm_prediction('<dBR/dq2>(B+->Kmumu)',q2min=q2_min,q2max=q2_max)*(q2_max-q2_min)

![Screenshot from 2023-06-14 18-34-44](https://github.com/flav-io/flavio/assets/17452912/1c3c90b2-ae8a-426b-b903-94864acf8af6)

**Comment by peterstangl** [comment]
Hi @gkaratha,

as far as I know, the currently implemented B->K form factors are different in flavio and EOS, which probably explains the differences in the plot. For results that go into a publication, it might be useful to update the the B->K form factors in flavio. This is on my todo list, but I'm not sure when I will find time to do it.

Apart from this, you should consider the theoretical uncertainties that are provided by flavio and EOS. This would also give you an indication, how large the differences actually are.

Please also note that the q2 range you are plotting contains narrow charmonium resonances that are not included in the plotted theory predictions. So these predictions are not meaningful around these resonances and just summing all the bins will not give you the total branching fraction. You might want to take a look at https://github.com/flav-io/flavio/issues/198, where one of your CMS colleagues has asked about this. Also note the comment there about the high q2 region, where the theory predictions are only meaningful for sufficiently wide bins (but not for the rather narrow bins shown in your plot).

**Comment by mreboud** [comment]
I agree with @peterstangl, the difference is mostly due to the default values used for the local form factors in EOS and flavio. I gave some EOS related details here https://github.com/eos/eos/issues/650.

**Comment by gkaratha** [comment]
Dear @peterstangl , @mreboud ,

Thanks a lot for the detailed answers! If the update of the form factors is straightforward, I can do it and make a PR. Or I can leave it as is and if you update them before we freeze the documents, I will update the plot accordingly. In the analysis we do not measure the BF in q2 close to the resonances and we plan to add uncertainties (for flavio are there). The plot is just to highlight the issue. Thanks a lot again.

Best regards,
George Karathanasis
